src/c4/Termination_Detector.cc

```diff
@@ -54,16 +54,18 @@ void Termination_Detector::init() {
 }
 
 //----------------------------------------------------------------------------//
-Termination_Detector::~Termination_Detector() {
 #ifdef PRINTF_DEBUG
+Termination_Detector::~Termination_Detector() {
   cout << pid_ << '/' << tag_ << ": destroyed" << endl;
-#endif
 }
+#else
+Termination_Detector::~Termination_Detector() = default;
+#endif
 
 //----------------------------------------------------------------------------//
 //! Return \c true if we have terminated; \c false otherwise.
 bool Termination_Detector::is_terminated() {
-  static unsigned buffer[3];
+  static std::array<unsigned, 3> buffer;
```

Comment:
Is 3 considered "magic number" code smell in this context, or is it okay?

Reply:
Created #795 to address this.